docs/application/native/guides/machine-learning/machine-learning-inference.md

```diff
@@ -298,6 +298,120 @@ Followings are the available elements:
     char pipeline[] = "videotestsrc is-live=true ! videoconvert ! tensor_converter ! output-selector name=outs outs.src_0 ! tensor_sink name=sink0 async=false outs.src_1 ! tensor_sink name=sink1 async=false"
     ```
 
+- **If**
+
+    The if (tensor_if) element allows creating conditional branches based on tensor values. For example, you may skip frames if there is no object detected with high confidence. The input and output stream data type is either `other/tensor` or `other/tensors`.
+
+    - **Properties**
+      - `compared-value`: Specifies the compared value and is represented as operand 1 from input tensors.
+        - A_VALUE: Decided based on a single scalar value.
+	    - TENSOR_AVERAGE_VALUE: Decided based on an average value of a specific tensor.
+	    - CUSTOM: Decided based on a user-defined callback.
+      - `compared-value-option`: Specifies an element of the nth tensor or you can pick one from the tensors.
+	    - [C][W][H][B],n (used for A_VALUE of the compared-value. e.g., 0:1:2:3,0 means [0][1][2][3] value of first tensor)
+        - nth tensor (used for TENSOR_AVERAGE_VALUE of the compared-value to specify which tensor are used.)
+      - `supplied-values`: Specifies the supplied value (SV) from the user.
+        - SV
+        - SV1, SV2 (used for RANGE operators)
+      - `operator`: Specifies a comparison operator which is used for comparing the compared-value (CV) and the supplied-value (SV).
+	    - EQ: Check if CV == SV
+	    - NEQ: Check if CV != SV
+	    - GT: Check if CV > SV
+	    - GE: Check if CV >= SV
+	    - LT: Check if CV < SV
+	    - LE: Check if CV <= SV
+	    - RANGE_INCLUSIVE: Check if SV1 <= CV and CV <= SV2
+	    - RANGE_EXCLUSIVE: Check if SV1 < CV and CV < SV2
+	    - NOT_IN_RANGE_INCLUSIVE: Check if CV < SV1 or SV2 < CV
+	    - NOT_IN_RANGE_EXCLUSIVE: Check if CV <= SV1 or SV2 <= CV
+      - `then`: Specifies the action, if the result is TRUE.
+	    - PASSTHROUGH: Does not let you make changes to the buffers. Buffers are pushed straight through.
+	    - SKIP: Does not let you generate the output frame (frame skip).
+        - TENSORPICK: Lets you choose the nth tensor among the input tensors.
+          ```
+          [ tensor 0 ]
+          [ tensor 1 ]  ->   tensor if    ->    [ tensor 0 ]
+          [ tensor 2 ]    (tensorpick 0,2)      [ tensor 2 ]
+          input tensors                         output tensors
+          ```
+      - `then-option`: Specifies the option for TRUE action.
+	    - nth tensor, nth tensor, (used for TENSORPICK option. e.g., then-option=0,2 means that tensor 0 and tensor 2 are selected as output tensors among the input tensors.)
```

Comment:
Can you please check this.

How about this ? Please verify for technical accuracy, then only incorporate.
```suggestion
          - nth tensor: used for TENSORPICK option, for example,  `then-option`=0,2 means tensor 0 and tensor 2 are selected as output tensors among the input tensors.

Reply:
Looks better. Updated.